test/migration_test.rb

```diff
@@ -229,5 +229,16 @@ def test_status_without_pending_migrations
       assert_output(output) { Mongoid::Migrator.status(MIGRATIONS_ROOT + "/valid") }
     end
 
+    def test_hook_after_migration
+      buffer = ""
+      Mongoid::Migration.after_migrate = Proc.new {
+        |output, name|
+        buffer = output
+      }
+      Mongoid::Migrator.up(MIGRATIONS_ROOT + "/other_valid")
+
+      assert_match(/^==  AddOtherPlanSurveySchema: migrating =======================================\n==  AddOtherPlanSurveySchema: migrated \(.+s\) ==============================\n\n$/, buffer)
```

Comment:
You actually want to use `\A` and `\z` instead of `^` and `$` here (https://stackoverflow.com/questions/577653/difference-between-a-z-and-in-ruby-regular-expressions), otherwise there could be all kind of other stuff in this string and the test would still pass as long as it's separated by a newline.